apps/omg_child_chain/test/omg_child_chain/integration/fee_server_test.exs

```diff
@@ -0,0 +1,155 @@
+      Process.sleep(500)
+    end)
+
+    %{fee_file: file_name}
+  end
+
+  describe "fees in effect" do
+    test "corrupted file does not make server crash", %{fee_file: file_name} do
+      {:started, _log, exit_fn} = start_fee_server()
+
+      default_fees = %{@eth => 0}
+      new_fee = 5
+
+      assert {:ok, default_fees} == FeeServer.transaction_fees()
+
+      # corrupt file, refresh, check fees did not change
+      overwrite_fee_file(file_name, "[not a json]")
+      assert refresh_fees() =~ ~r/\[error\].*Unable to update fees/
+
+      assert {:ok, default_fees} == FeeServer.transaction_fees()
+      assert server_alive?()
+
+      # fix file, reload, check changes applied
+      overwrite_fee_file(file_name, %{@eth_hex => new_fee})
+      refresh_fees()
+
+      assert {:ok, %{@eth => new_fee}} == FeeServer.transaction_fees()
+      assert server_alive?()
+
+      exit_fn.()
+    end
+
+    test "starting with corrupted file makes server die", %{fee_file: file_name} do
+      overwrite_fee_file(file_name, "[not a json]")
+      {:died, log} = start_fee_server()
+
+      assert log =~ ~r/\[error\].*Unable to update fees/
+      assert false == server_alive?()
+    end
+  end
+
+  describe "fees ignored" do
+    setup do
+      Application.put_env(:omg_child_chain, :ignore_fees, true)
+      {:started, log, exit_fn} = start_fee_server()
+      assert log =~ "ignored"
+
+      on_exit(fn ->
+        exit_fn.()
+        Application.put_env(:omg_child_chain, :ignore_fees, false)
+      end)
+
+      :ok
+    end
+
+    test "fee server ignores file updates" do
+      assert {:ok, :ignore} == FeeServer.transaction_fees()
+
+      assert refresh_fees() =~ "Updates takes no effect"
+
+      assert {:ok, :ignore} == FeeServer.transaction_fees()
+      assert server_alive?()
+    end
+  end
+
+  defp start_fee_server do
+    log = capture_log(fn -> GenServer.start(FeeServer, [], name: TestFeeServer) end)
+
+    case GenServer.whereis(TestFeeServer) do
+      pid when is_pid(pid) ->
+        # switch of internal timer to don't interfere with tests
+        if tref = Keyword.get(:sys.get_state(TestFeeServer), :tref) do
+          {:ok, :cancel} = :timer.cancel(tref)
+        end
+
+        {:started, log, fn -> GenServer.stop(pid) end}
+
+      nil ->
+        {:died, log}
+    end
+  end
+
+  defp refresh_fees do
+    pid = GenServer.whereis(TestFeeServer)
+
+    capture_log(fn ->
+      Process.send(pid, :update_fee_spec, [])
+      # handle_info is async - we need to wait it executes to get message logged
+      Process.sleep(100)
+    end)
+  end
+
+  defp server_alive? do
+    case GenServer.whereis(TestFeeServer) do
+      pid when is_pid(pid) ->
+        Process.alive?(pid)
+
+      _ ->
+        false
+    end
+  end
+
+  defp overwrite_fee_file(file, content) do
+    # file modification date is in seconds
+    Process.sleep(1000)
+    {:ok, _, ^file} = TestHelper.write_fee_file(content, file)
```

Comment:
nice! cool for the explicit TestHelper call!

Reply:
Is it :+1:  or :-1: :question:

Reply:
it's great, I hate imports :D